### ISSUE TITLE
Declaring Objects with Const

### ISSUE THREAD

**Issue by vijayperlakota** [nzakas/understandinges6#231]
I was executing the code in chrome console given in section 'Declaring Objects with Const'. It did not throw err for me. Is there any specific/prior things to do before executing the code?
![screenshot from 2016-01-30 13 36 49](https://cloud.githubusercontent.com/assets/5469709/12694534/f751fde0-c756-11e5-908b-cf9f84d0c12c.png)

const person = {
    name: "Nicholas"
};

// works
person.name = "Greg";

// throws an error
person = {
    name: "Greg"
};

**Comment by getify** [comment]
this is correct behavior. `const` only specifies a constant assignment, not an immutable value. to make an object value immutable, you need to also use `Object.freeze(..)` on it, though note that this is shallow and thus only makes the top level of that object's properties frozen from re-assignment.

**Comment by nzakas** [comment]
@getify I think you missed that the reassignment to `person` should throw an error but isn't in the browser. 

@vijayperlakota the book is correct in its description, but not all browsers implement everything correctly right now. Chrome had implemented `const` as a synonym for `var` early on. Chrome 48 still doesn't have this implemented correctly. See https://kangax.github.io/compat-table/es6/

Chrome 49 should have it correct.

**Comment by getify** [comment]
Reassignment will not throw an error in non-strict mode, it will just silently fail.

**Comment by nzakas** [comment]
Hmm, is that true? It looks like it's supposed to throw an error in both strict and non-strict modes:
https://tc39.github.io/ecma262/#sec-blockdeclarationinstantiation

CreateImmutableBinding (https://tc39.github.io/ecma262/#sec-environment-records):

> Create a new but uninitialized immutable binding in an Environment Record. The String value N is the text of the bound name. If S is true then attempts to access the value of the binding before it is initialized or set it after it has been initialized will always throw an exception, regardless of the strict mode setting of operations that reference that binding. S is an optional parameter that defaults to false

It looks like `S` is true whenever a lexical declaration is a constant. Or am I missing something?

**Comment by rwaldron** [comment]
@nzakas @getify 

tl;dr: `const` reassignment throws in both non-strict & strict code. 


- https://tc39.github.io/ecma262/#sec-globaldeclarationinstantiation (Step 16.b.i)
- https://tc39.github.io/ecma262/#sec-global-environment-records-createimmutablebinding-n-s (step 3)

**Comment by nzakas** [comment]
Thanks @rwaldron <3

**Comment by rwaldron** [comment]
Anytime!
